### PR TITLE
update gl-native to 10.5.0-beta.1, common to v21.3.0-beta.2

### PR DESCRIPTION
**Pull request by ank27**
### Summary of changes

update gl-native to 10.5.0-beta.1, common to v21.3.0-beta.2

## Breaking changes ⚠️
* Map projection API moved from Map to Style, in order to allow specifying the map projection in the style. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))
* Automatic transition between the globe and mercator projection updated to appear visually more subtle. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))

## Features ✨ and improvements 🏁
* Map render call optimized further by further reducing computational overhead. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))
* Layer properties transitions performance improved if the layer is transitioning to the same constant value or if transitioning from/to data-driven property. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))
* New line layer paint property introduced: '{"line-trim-offset", [trim-start, trim-end]}', to take the line trim-off percentage range based on the whole line range [0.0, 1.0]. The property will only be effective when 'line-gradient' property is set. The line part between [trim-start, trim-end] will be marked as transparent to make a line gradient a vanishing effect. If either 'trim-start' or 'trim-end' offset is out of valid range, the default range [0.0, 0.0] will be set. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))
* Globe view controls revamped for more intuitive interaction with touch controls. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))
* OfflineRegion::getStatus() API added to get the completion status and the local size of the existing legacy offline regions. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))

## Bug fixes 🐞
* Dispatched in-flight events will not be delivered if 'unsubscribe' is called before an event is delivered. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))
* Transitions between globe and mercator projection do not cull tiles incorrectly anymore. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))
* Map LOD disabled for camera pitch less than 30 degrees to avoid map content missing on maps with insets. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))
* Terrain-related camera issues fixed, previously making it appear under terrain, or incorrectly repositioned after exaggeration change. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))
* Terrain rendering disabled on GPUs not supporting Vertex Texture Fetch. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))
* Fixed a bug that occasionally prevented symbols from loading. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))

## Dependencies
* Bump gl-native to v10.5.0-beta.1, mapbox-common to v21.3.0-beta.2. ([#1244](https://github.com/mapbox/mapbox-maps-android/pull/1244))

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [x] If this PR is a `v10.5` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.5` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: LICENSE.md
Comment (kiryldz) [inline LICENSE.md:31]: @ank27 do ew have understanding where does this come from? and if it's expected. Datastore / coroutines - is this what we need in SDK?

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt
Comment (kiryldz) [inline sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:1694]: ok with null here as this will be revisited later

File: LICENSE.md
Comment (ank27) [reply LICENSE.md:31]: Its coming from data store implementation from common's settings api for i18n. its expected and we do need these.

datastore internally uses coroutines, which means this is also required.

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt
Comment (kiryldz) [inline sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:1712]: ok with mercator here as this will be revisited later.
however we need to make sure that tests don't fail or (if they do) - comment them out for now, I'll handle that after this PR is merged

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt
Comment (kiryldz) [inline sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:1748]: not sure I like default ktlint formatting here but up to you 😄

File: sdk/src/main/java/com/mapbox/maps/Style.kt
Comment (kiryldz) [inline sdk/src/main/java/com/mapbox/maps/Style.kt:892]: same as https://github.com/mapbox/mapbox-maps-android/pull/1244/files#r840679562

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt
Comment (ank27) [reply sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:1694]: `null` value means it will always return `Mercator`, but we can discuss what we need to do about these methods in comment https://github.com/mapbox/mapbox-maps-android/pull/1244#discussion_r840677196

**Review by kiryldz — APPROVED** [review]
dropping 🟢 in order to unblock merging, several concerns / questions:
 - revapi needs to be run as public api changed
 - some concerns about licenses, needs extra validation
 - some other minor points I stated as comments

File: LICENSE.md
Comment (kiryldz) [reply LICENSE.md:31]: thanks for clarification, that's good it's expected although I'm curious how it will affect binary size, we need to keep an eye on that

File: sdk/src/androidTest/java/com/mapbox/maps/GlobeIntegrationTest.kt
Comment (ank27) [inline sdk/src/androidTest/java/com/mapbox/maps/GlobeIntegrationTest.kt:45]: these will be fixed when StyleProjection work done as a tailwork.

**Review by alexander-kulikovskii — APPROVED** [review]

**Comment by ank27** [comment]
somehow `verify-api-metalava` job is failing. any idea @mapbox/android ?

File: CHANGELOG.md
Comment (ank27) [reply CHANGELOG.md]: moved that to feature.

**Review by pengdev — APPROVED** [review]

**Pull request merged**
